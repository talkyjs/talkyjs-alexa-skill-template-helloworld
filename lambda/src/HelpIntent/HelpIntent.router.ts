import { Router } from "@talkyjs/core";

import { HelpIntentScript } from './HelpIntent.speech'


export const HelpIntentRouter: Router = {
    requestType: "IntentRequest",
    intentName: "AMAZON.HelpIntent",
    handler: async (handlerInput) => {
        
        
        const script = new HelpIntentScript(handlerInput)
        return script
            .createResponseBuilder()
            .getResponse();
        
        
    }
}

export default HelpIntentRouter