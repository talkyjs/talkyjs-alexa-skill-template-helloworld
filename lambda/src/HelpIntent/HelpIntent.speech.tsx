import React from 'react';
import { SpeechScriptJSX } from '@talkyjs/ssml'

export class HelpIntentScript extends SpeechScriptJSX {
    speech() {
        return (
            <speak>
                <p>Hello! It's a nice development. How are you?</p>
            </speak>
        )
    }
    
    reprompt() {
        return (
            <speak>
                <p>How are you?</p>
            </speak>
        )
    }
    
}