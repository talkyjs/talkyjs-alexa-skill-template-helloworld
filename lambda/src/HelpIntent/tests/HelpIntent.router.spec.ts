import { HandlerInputCreator } from '@ask-utils/test';
import { RequestHandlerFactory } from '@talkyjs/core';
import { RequestHandler } from 'ask-sdk-core';
import { HelpIntentRouter } from '../HelpIntent.router'

describe('HelpIntentRouter', () => {
  let handler: RequestHandler;
  beforeEach(() => {
    handler = RequestHandlerFactory.create(HelpIntentRouter);
  });
  describe('canHandle', () => {
    it('should return false when given a not LaunchRequest', async () => {
      const handlerInput = new HandlerInputCreator().createLaunchRequest();
      await expect(handler.canHandle(handlerInput)).resolves.toEqual(false);
    });
    
    it('should return false when given a not IntentRequest', async () => {
      const handlerInput = new HandlerInputCreator().createIntentRequest({
        name: "AMAZON.HelpIntent",
        confirmationStatus: 'NONE'
      });
      await expect(handler.canHandle(handlerInput)).resolves.toMatchSnapshot();
    });
    
  });
  describe('handle', () => {
    it('should match snapshot', async () => {
      const handlerInput = new HandlerInputCreator().createLaunchRequest();
      await expect(handler.handle(handlerInput)).resolves.toMatchSnapshot();
    });
  });
});
