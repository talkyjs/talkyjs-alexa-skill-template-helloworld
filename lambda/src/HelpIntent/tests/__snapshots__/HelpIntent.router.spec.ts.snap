// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpIntentRouter canHandle should return false when given a not IntentRequest 1`] = `true`;

exports[`HelpIntentRouter handle should match snapshot 1`] = `
Object {
  "outputSpeech": Object {
    "ssml": "<speak><p>Hello! It's a nice development. How are you?</p></speak>",
    "type": "SSML",
  },
  "reprompt": Object {
    "outputSpeech": Object {
      "ssml": "<speak><p>How are you?</p></speak>",
      "type": "SSML",
    },
  },
  "shouldEndSession": false,
}
`;
